src/core/geometry/qgscompoundcurve.cpp

```diff
@@ -388,6 +388,22 @@ bool QgsCompoundCurve::isEmpty() const
   return true;
 }
 
+bool QgsCompoundCurve::isValid( QString &error SIP_OUT, int flags ) const
+{
+  if ( mCurves.isEmpty() )
+    return true;
+
+  for ( QgsCurve *curve : mCurves )
+  {
+    if ( !curve->isValid( error, flags ) )
+    {
+      error = QString( "At least one curve is not valid." );
```

Comment:
Should this be either `tr` or `QStringLiteral`

Reply:
And wouldn't it be better to also include the actual reason the curve is failing? ie "Curve 1: Geometry has less than 2 points and is not empty."